File: FamilyTree/Models/Tree/ConnectNodesRequest.cs
using System.ComponentModel.DataAnnotations;

namespace FamilyTree.Models
{
    public class ConnectNodesRequest
    {
        [Required]
        public int TreeId { get; set; }
        public int ChildId { get; set; }
        public int FirstParentId {get; set;}
        public int? SecondParentId {get; set;}
    }
}

File: FamilyTree/ClientApp/src/components/tree/API/connectNodes/connectNodesRequest.ts
import axios from "axios";
import { baseURL } from "../../../../helpers/apiHelpers";
import { TreeAPI } from "./../utils/TreeModel";

export type ConnectNodesRequestData = {
  treeId: number;
  childId: number;
  firstParentId: number;
  secondParentId?: number;
};

export type ConnectNodesResponse = TreeAPI;

export const connectNodesRequest = (data: ConnectNodesRequestData) => {
  return axios.post(`${baseURL}/tree/node/connect`, data);
};
